src/engine/openspaceengine.cpp

```diff
@@ -497,22 +499,124 @@ void OpenSpaceEngine::initialize() {
+    windowWrapper().setSynchronization(false);
+    OnExit(
+        [this]() {
+            windowWrapper().setSynchronization(true);
+            windowWrapper().setBarrier(true);
+        }
+    );
+    
+    // Run start up scripts
+    try {
+        runPreInitializationScripts(scenePath);
+    }
+    catch (const ghoul::RuntimeError& e) {
+        LERRORC(e.component, e.message);
+    }
+
+
+    Scene* scene;
+    try {
+        scene = _sceneManager->loadScene(scenePath);
+    } catch (const ghoul::FileNotFoundError& e) {
+        LERRORC(e.component, e.message);
+        return;
+    } catch (const Scene::InvalidSceneError& e) {
+        LERRORC(e.component, e.message);
+        return;
+    } catch (const ghoul::RuntimeError& e) {
+        LERRORC(e.component, e.message);
+        return;
+    }
+
+    Scene* previousScene = _renderEngine->scene();
+    if (previousScene) {
+        _syncEngine->removeSyncables(Time::ref().getSyncables());
+        _syncEngine->removeSyncables(_renderEngine->getSyncables());
+        _syncEngine->removeSyncable(_scriptEngine.get());
+
+        _renderEngine->setScene(nullptr);
+        _renderEngine->setCamera(nullptr);
+        _sceneManager->unloadScene(*previousScene);
+    }
+
+    // Initialize the RenderEngine
+    _renderEngine->setScene(scene);
+    _renderEngine->setCamera(scene->camera());
     _renderEngine->setGlobalBlackOutFactor(0.0);
     _renderEngine->startFading(1, 3.0);
-    
+
+    scene->initialize();
+    _interactionHandler->setCamera(scene->camera());
+
+    try {
+        runPostInitializationScripts(scenePath);
+    }
+    catch (const ghoul::RuntimeError& e) {
+        LFATALC(e.component, e.message);
+    }
+
+    // Write keyboard documentation.
+    const std::string KeyboardShortcutsType =
+        ConfigurationManager::KeyKeyboardShortcuts + "." +
+        ConfigurationManager::PartType;
+
+    const std::string KeyboardShortcutsFile =
+        ConfigurationManager::KeyKeyboardShortcuts + "." +
+        ConfigurationManager::PartFile;
+
+    std::string type, file;
+    bool hasType = configurationManager().getValue(KeyboardShortcutsType, type);
+    bool hasFile = configurationManager().getValue(KeyboardShortcutsFile, file);
+
+    if (hasType && hasFile) {
+        interactionHandler().writeKeyboardDocumentation(type, file);
+    }
+
+    // If a PropertyDocumentationFile was specified, generate it now.
+    const std::string KeyPropertyDocumentationType =
+        ConfigurationManager::KeyPropertyDocumentation + '.' +
+        ConfigurationManager::PartType;
+
+    const std::string KeyPropertyDocumentationFile =
+        ConfigurationManager::KeyPropertyDocumentation + '.' +
+        ConfigurationManager::PartFile;
+
+    hasType = configurationManager().hasKey(KeyPropertyDocumentationType);
+    hasFile = configurationManager().hasKey(KeyPropertyDocumentationFile);
+
+    if (hasType && hasFile) {
+        std::string propertyDocumentationType;
```

Comment:
Can use `configurationManager().value<std::string>` since you already checked that the key exists